src/components/PageContent/PageContent.astro

```diff
@@ -1,49 +1,194 @@
-    </nav>
-    <slot />
-
-    <slot name="after-content" />
-    {whatsNext && <WhatsNext content={whatsNext} />}
-  </section>
-  <nav class="block sm:hidden">
-    <MoreMenu editHref={githubEditUrl} />
-  </nav>
+<article id="article">
+  <ContentObserver client:media="(min-width: 50em)" />
+  <Overview title={frontmatter.title} />
+  <span class="sm:hidden">
+    <TableOfContents client:media="(max-width: 50em)" />
+  </span>
+  <slot />
 </article>
 
-<style>
-  .content {
-    padding: 0;
-    /* max-width: 75ch; */
-    width: 100%;
-    height: 100%;
+<style is:inline>
+  article {
+    position: relative;
+  }
+
+  article * {
+    max-width: 100%;
+    margin-bottom: 0;
+  }
+
+  article :is(label) {
     display: flex;
-    flex-direction: column;
+    align-items: center;
+    gap: var(--space-2x);
+  }
+
+  article :where(section) > :not(section),
+  article > :not(section) {
+    margin-top: var(--space-5x);
+  }
+
+  article :is(h1, h2) {
+    top: var(--theme-navbar-height, 2rem);
+    z-index: 3;
+  }
+
+  article :is(h2) {
+    position: sticky;
+    background: white;
+    font-size: 28px;
+    padding: var(--space-6x) 0;
+    margin-bottom: calc(-1 * var(--space-5x));
+  }
+
+  article :is(h3, h4) {
+    padding-top: var(--space-8x);
   }
 
-  .content > section {
-    margin-bottom: 4rem;
+  article :is(h3, h4, li) {
+    margin-top: 0;
   }
 
-  .block {
-    display: block;
+  article > :first-child {
+    margin-top: 0;
+  }
+
+  article :is(ul) {
+    padding-left: var(--space-4x);
+  }
+
+  article :is(ul, ol) {
+    margin-top: var(--space-3x);
+  }
+
+  article :is(ul, ol) > li > :is(ul, ol) {
+    margin-top: var(--space-2x);
+  }
+
+  article :is(ul, ol) > * + * {
+    margin-top: var(--space-2x);
+  }
+
+  article li > :is(p, pre, blockquote):not(:first-child) {
+    margin-top: var(--space-4x);
+  }
+
+  article :is(ol) {
+    padding-left: var(--space-8x);
+  }
+
+  article ::marker {
+    font-weight: bold;
+    color: var(--theme-text-light);
+  }
+
+  article iframe {
+    width: 100%;
+    height: auto;
+    aspect-ratio: 16 / 9;
+  }
+
+  section:target {
+    scroll-margin-top: var(--theme-navbar-height, 2rem);
+  }
+
+  section > section:target {
+    scroll-margin-top: calc(var(--theme-navbar-height, 2rem) + 5rem);
+  }
+
+  .sm\:hidden {
+    display: initial;
   }
 
   @media (min-width: 50em) {
     .sm\:hidden {
       display: none;
     }
+    article :is(h1, h2) {
+      position: sticky;
+      background: white;
+      border-bottom: 2px solid var(--blue-200, #dfe7fb);
+      top: 0;
+      padding: var(--space-8x) 0;
+    }
+
+    article :is(h1) {
+      margin-top: 0;
+    }
+
+    article :is(h2) {
+      font-size: 32px;
+      margin-bottom: 0;
+    }
+
+    article :is(h4) {
+      font-size: 32px;
+    }
+
+    section:target {
+      scroll-margin-top: 0;
+    }
+
+    section > section:target {
+      scroll-margin-top: 6rem;
+    }
   }
 </style>
+
+<script>
```

Comment:
I think that we can probably do this at the markdown level:

We could use the `remarkHeadings` plugin to modify the generated html: https://docs.astro.build/en/guides/markdown-content/#heading-ids-and-plugins

Or we could specify custom components to be used for the headings 
https://docs.astro.build/en/guides/integrations-guide/mdx/#customcomponentnames

Have a look at those options and we can chat about it some time next week